specification/design-principles.md

```diff
@@ -0,0 +1,5 @@
+# Design Principles
+
+## Overview
+
+// TODO: this is a stub.
```

Comment:
I think it's a good idea to mention some high-level design philosophies, for instance:

- no/low dependency evaluation API
- vendor neutrality
- extensibility concepts (hooks, etc)
- first class integrations (OTel, for example)